packages/js/client/src/__tests__/Web3ApiClient.spec.ts

```diff
@@ -121,6 +122,56 @@ describe("Web3ApiClient", () => {
     ]);
   });
 
+  it("invoke with decode false/true works as expected", async () => {
+    const api = await buildAndDeployApi(
+      `${GetPathToTestApis()}/simple-storage`,
+      ipfsProvider,
+      ensAddress
+    );
+
+    const ensUri = `ens/testnet/${api.ensDomain}`;
+    const client = await getClient();
+
+    // The decode option is defaulted to true
+    {
+      const result = await client.invoke<string>({
+        uri: ensUri,
+        module: "mutation",
+        method: "deployContract",
+        input: {
+          connection: {
+            networkNameOrChainId: "testnet"
+          }
+        },
+      });
+
+      expect(result.error).toBeFalsy();
+      expect(result.data).toBeTruthy();
+      expect(typeof result.data).toBe("string");
+      expect(result.data).toContain("0x");
+    }
+
+    // When decode is set to false, an ArrayBuffer is returned
+    {
+      const result = await client.invoke({
+        uri: ensUri,
+        module: "mutation",
+        method: "deployContract",
+        input: {
+          connection: {
+            networkNameOrChainId: "testnet"
+          }
+        },
+        decode: false
```

Comment:
It's usually not a great practice to have to set params to false to override behavior
Defaults should be false by default

Reply:
Great point, as discussed I'll change this to `noDecode` instead.